backends/arm/test/tester/arm_tester.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2024 Arm Limited and/or its affiliates.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from enum import Enum
+from typing import Optional, Tuple
+
+import torch
+from executorch.backends.arm.arm_backend import (
+    generate_ethosu_compile_spec,
+    generate_tosa_compile_spec,
+)
+
+from executorch.backends.arm.arm_partitioner import ArmPartitioner
+
+from executorch.backends.arm.test.tosautil.tosa_test_utils import (
+    TosaProfile,
+    TosaTestUtils,
+)
+
+from executorch.backends.xnnpack.test.tester import Tester
+from executorch.backends.xnnpack.test.tester.tester import (
+    Export,
+    Partition,
+    Quantize,
+    ToEdge,
+)
+
+from executorch.exir import EdgeCompileConfig
+from torch.ao.quantization.quantizer.xnnpack_quantizer import (
+    get_symmetric_quantization_config,
+    XNNPACKQuantizer,
+)
+
+
+class ArmBackendSelector(Enum):
+    TOSA = "tosa"
+    ETHOS_U55 = "ethos-u55"
+
+
+class ArmTester(Tester):
+    def __init__(
+        self,
+        model: torch.nn.Module,
+        inputs: Tuple[torch.Tensor],
+        backend: ArmBackendSelector = ArmBackendSelector.TOSA,
+        profile: TosaProfile = TosaProfile.BI,
+    ):
+        """
+        Args:
+            model (torch.nn.Module): The model to test
+            inputs (Tuple[torch.Tensor]): The inputs to the model
+            backend (ArmBackendSelector): The backend to use. E.g. TOSA or
+                ETHOS_U55.
+                TOSA: Lower to TOSA and test numerical correctness compared to
+                    torch reference.
+                ETHOS_U55: Lower to TOSA, then let Vela compile. Only
+                    functional test, no numerical checks.
+            profile (TosaProfile): The TOSA profile to use. Either
+                TosaProfile.BI or TosaProfile.MI
+        """
+        self.tosa_test_util = None
+        if backend == ArmBackendSelector.TOSA:
+            self.tosa_test_util = TosaTestUtils(profile=profile)
+            # The spec below tiggers arm_backend.py to output two files:
+            #   1) output.tosa
+            #   2) desc.json
+            # Saved on disk in self.tosa_test_util.intermediate_path
+            self.compile_spec = generate_tosa_compile_spec(
+                self.tosa_test_util.intermediate_path
+            )
+        elif backend == ArmBackendSelector.ETHOS_U55:
+            self.compile_spec = generate_ethosu_compile_spec("ethos-u55-128")
+        else:
+            raise ValueError(f"Unknown backend: {backend}")
+        super().__init__(model, inputs)
+
+    def quantize(self, quantize_stage: Optional[Quantize] = None):
+        if quantize_stage is None:
+            # Using the XNNPACKQuantizer for now
+            quantize_stage = Quantize(
+                XNNPACKQuantizer(),
```

Comment:
@freddan80 - can we just fork this and make it TOSAQuantizer - again just copy-paste for now then we can diverge. Also our plan for XNNPACKQuantizer to live inside backends/xnnpack so perhaps when you fork you can start with copying TOSAQuantizer in backends/arm dir.

Reply:
We have patch for that, but it's stuck at the moment. Would it be possible to merge this first?